### PR TITLE
fix: clarity: no user supplied rcmgr limits of 0

### PR DESCRIPTION
**Pull request by BigLep**
The goal of this commit is to make clear that we don't support user-supplied limits of 0 currently. Apply (https://github.com/libp2p/go-libp2p/blob/master/p2p/host/resource-manager/limit.go#L102 ) overrides any 0 value.   With the current code, a zero-limit in userSuppliedOverrideLimitConfig will be overridden by the computed default limits.

I want to be able to do:

```
defaultComputedLimitConfig.Apply(userSuppliedOverrideLimitConfig)
```

but that won't override any of the computed defaults.

I'd like to see this fixed, but this at least make clear the situation.

### CODE REVIEW

File: core/node/libp2p/rcmgr.go [hunk 1]
```diff
@@ -61,10 +62,15 @@ func ResourceManager(cfg config.SwarmConfig) interface{} {
...
 			if cfg.ResourceMgr.Limits != nil {
-				l := *cfg.ResourceMgr.Limits
-				// This effectively overrides the computed default LimitConfig with any vlues from cfg.ResourceMgr.Limits
-				l.Apply(limitConfig)
-				limitConfig = l
+				userSuppliedOverrideLimitConfig := *cfg.ResourceMgr.Limits
+				// This effectively overrides the computed default LimitConfig with any non-zero values from cfg.ResourceMgr.Limits.
+				// Because of how how Apply works, any 0 value for a user supplied override 
+				// will be overriden with a computed default value.
+				// There currently isn't a way for a user to to supply a 0-value override.
```
Comment (ajnavarro) [inline core/node/libp2p/rcmgr.go:69]:
```suggestion
				// There currently isn't a way for a user to supply a 0-value override.
```

**Review by ajnavarro — APPROVED** [review]

File: core/node/libp2p/rcmgr.go (hunk 1 above)
Comment (lidel) [inline core/node/libp2p/rcmgr.go:69]: I think we should add this paragraph to https://github.com/ipfs/kubo/blob/master/docs/config.md#swarmresourcemgrlimits, so users know "0 == autoscaled default"

Comment (BigLep) [reply core/node/libp2p/rcmgr.go]: Good call @lidel .  I updated the docs.

**Comment by BigLep** [comment]
I think we're good to merge now @lidel , but will let you check.

**Comment by ajnavarro** [comment]
@BigLep You need to format the code to make Go checks pass.

**Comment by BigLep** [comment]
Doh - thanks @ajnavarro .  I guess that's what I get for making Go code changes in the Github UI.  It looks like I'm good now...

**Review by lidel — APPROVED** [review]
I took this PR for a spin and agree we are better with it in 0.18. Merging. :+1:  


Some thoughts:
- Without this PR we would have very bad time migrating and cleaning up user configs in 0.19 when go-libp2p has explicit `0` support. Good call @BigLep 
- UX  of doing anything with ResourceMgr via CLI is pure torture, but since we have autoscaling I hope not many people will have to touch it. 
-  I've found two bugs and filled  https://github.com/ipfs/kubo/issues/9576 and https://github.com/ipfs/kubo/issues/9577  but none of them should block 0.18, fine to fix them later.

**Pull request merged**